scripts/apps/authoring/attachments/attachments.js

```diff
@@ -1,97 +1,67 @@
 import './attachments.scss';
 import {gettext} from 'core/utils';
+import AttachmentsEditorDirective from './AttachmentsEditorDirective';
+
+export {attachments} from './reducer';
+export {initAttachments} from './actions';
+
+import {
+    removeFile,
+    selectFiles,
+    editFile,
+    saveFile,
+    closeEdit,
+    download,
+} from './actions';
+
+const mapStateToCtrl = (state) => ({
+    edit: state.attachments.diff,
+    file: state.attachments.file,
+    files: state.attachments.files,
+    maxSize: state.attachments.maxSize,
+    maxFiles: state.attachments.maxFiles,
+    isLocked: state.editor.isLocked,
+    isLockedByMe: state.editor.isLockedByMe,
+});
+
+const mapDispatchToCtrl = (dispatch) => ({
+    selectFiles: (files) => dispatch(selectFiles(files)),
+    removeFile: (file) => dispatch(removeFile(file)),
+    editFile: (file) => dispatch(editFile(file)),
+    saveFile: (file, diff) => dispatch(saveFile(file, diff)),
+    closeEdit: () => dispatch(closeEdit()),
+    download: (file) => dispatch(download(file)),
+});
 
 class AttachmentsController {
-    constructor($scope, $window, superdesk, attachments, notify, deployConfig, urls, lock) {
-        this.$scope = $scope;
-        this.$window = $window;
-        this.superdesk = superdesk;
-        this.attachments = attachments;
-        this.notify = notify;
-        this.urls = urls;
-        this.isLocked = lock.isLocked($scope.item);
-        this.isLockedByMe = lock.isLockedByMe($scope.item);
-
-        attachments.byItem($scope.item).then((files) => {
-            this.$scope.files = files;
-        }, () => {
-            this.$scope.files = [];
-        });
-
-        this.maxFiles = deployConfig.getSync('attachments_max_files');
-        this.maxSize = deployConfig.getSync('attachments_max_size');
-    }
-
-    selectFiles(files) {
-        if (Array.isArray(files) && files.length > 0 && !this.isLocked) {
-            if (files.length + this.$scope.files.length > this.maxFiles) {
-                this.notify.error(gettext('Sorry, too many files selected.'));
-                return;
+    constructor($scope) {
+        // connect
+        Object.assign(this, mapDispatchToCtrl($scope.store.dispatch));
```

Comment:
It isn't very clear how this works. Could you instead use a var like
```
const initialState = Object.assign(this, mapDispatchToCtrl($scope.store.dispatch));
```
and use that `initialState` where you need. I would also advise not to modify the original object by passing an empty object as a first argument to `Object.assign`. That way it's more clear what are the limits of influence of that variable.

Reply:
here it's populating controller methods, which are then called from template. it's there to avoid:
```
this.edit = (file) => dispatch(edit(file));
this.close = () => dispatch(close());
```

Reply:
This is way more readable to me. I don't think there's any point of dynamic assignment when it's only used in one place.
```
const {dispatch} = $scope.store;

this.edit = (file) => dispatch(edit(file));
this.close = () => dispatch(close());
// all the other methods
```

Reply:
I wanted to make some utils do handle it, something like in [redux-react](https://react-redux.js.org/api/connect#dispatch)